Failure when adding tag in the CI

Tag task fails with the error
```
Run VERSION=$(git describe --tags $(git rev-list --tags --max-count=1))
fatal: No names found, cannot describe anything.
Error: Process completed with exit code 128.
```

Fix issue in github actions pipeline

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -35,6 +35,8 @@ jobs:
     steps:
     - name: Check out repository
       uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
 
     - name: Set up Git
       run: |
@@ -44,10 +46,7 @@ jobs:
     - name: Create tag
       id: tag
       run: |
-        VERSION=$(git describe --tags $(git rev-list --tags --max-count=1))
-        if [ -z "$VERSION" ]; then
-          VERSION=0.0.0
-        fi
+        VERSION=$(git describe --tags $(git rev-list --tags --max-count=1) || echo "0.0.0")
         VERSION_BITS=(${VERSION//./ })
         MAJOR=${VERSION_BITS[0]}
         MINOR=${VERSION_BITS[1]}

--- a/README.md
+++ b/README.md
@@ -148,3 +148,9 @@ This project is licensed under the MIT License. See the LICENSE file for details
 
 Developed by [ChatGPT](https://chatgpt.com).
 
+---
+
+## CI Pipeline Tagging Mechanism
+
+The CI pipeline now includes a tagging mechanism that ensures the versioning starts from `0.0.0` if no tags are found. This prevents the pipeline from failing when no tags are present in the repository.
+

